django/template/base.py

```diff
@@ -204,19 +204,15 @@ def _render(self, context):
 
     def render(self, context):
         "Display stage -- can be called many times"
-        # Set context.template to the original template -- as opposed to
-        # extended or included templates -- during rendering. This may be
-        # used for accessing context.template.engine.
-        toplevel_render = context.template is None
-        if toplevel_render:
-            context.template = self
         context.render_context.push()
         try:
-            return self._render(context)
+            if context.template is None:
+                with context.bind_template(self):
+                    return self._render(context)
+            else:
+                return self._render(context)
```

Comment:
Ideally I would like to call `context.bind_template(self)` unconditionally.

That may become possible if we make the change described in https://github.com/django/django/pull/4161#issuecomment-75136193